### PR TITLE
Refactor school library code

### PR DESCRIPTION
**Pull request by ifeanyi305**
Hi reviewer in this PR my coding partner and I refactored the school library code following the principles of SOLID.
* S - Single-responsibility Principle
* O - Open-closed Principle
* L - Liskov Substitution Principle
* I - Interface Segregation Principle
* D - Dependency Inversion Principle
We also:
Made sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
Followed SOLID principles.
Payed special attention to the "Single-responsibility Principle".
Made sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,6 +96,12 @@ To run tests, run the following command:
 
 - GitHub: [Mohamed Talbi](https://github.com/Kweeka1/)
 
+👤 **Ifeanyi Oti**
+
+- GitHub: [@githubhandle](https://github.com/ifeanyi305)
+- Twitter: [@twitterhandle](https://twitter.com/OtiJoseph305)
+- LinkedIn: [LinkedIn](https://linkedin.com/in/oti-joseph-ifeanyi)
+
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```
Comment (Munsa1) [inline README.md:106]: - [ ] Kindly fix the broken link to the MIT license in the readme file as this is a requirement

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @ifeanyi305 @Kweeka1 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Passing Linters :heavy_check_mark: 
Correct GitHub Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md [hunk 2]
```diff
@@ -148,6 +154,6 @@ I would like to thank Microverse for project inspiration.
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
+This project is [MIT](./MIT.md) licensed.
```
Comment (cynthiainga) [inline README.md:157]: - [OPTIONAL] I think it would be better if you update your MIT.md file 🙃 😉

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ 🎉 👍 

Hi @Kweeka1 & @ifeanyi305, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**